app/src/main/kotlin/compiler/common/dfa/DfaWalk.kt

```diff
@@ -0,0 +1,13 @@
+package compiler.common.dfa
+
+interface DfaWalk<A, R> {
+    fun getAcceptingStateTypeOrNull(): R?
+
+    fun isAccepting(): Boolean {
```

Comment:
One shouldn't be able to override this method. We can replace it with an extension method.